src/librbd/internal.cc

```diff
@@ -3445,6 +3445,23 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
     return r;
   }
 
+  int flush_cache(ImageCtx *ictx)
```

Comment:
There is no much difference. And initially I did it using flush(), but after looking at the result it seemed to me it would be cleaner if I added flush_cache(), being consistent with invalidate_cache() (e.g. the caller does not need to acquire the locks, also counter names look similar).

There is some confusion with adding a new flush_cache counter though,  and actually I don't have a strong opinion here, so I would happy to do what you suggest. Should I abandon flush_cache() and call flush() instead?